src/codemagic/tools/_app_store_connect/arguments.py

```diff
@@ -357,6 +353,17 @@ def _apply_type(cls, non_typed_value: str) -> List[BetaBuildInfo]:
 
             return beta_build_infos
 
+    class DeviceUdidsArgument(cli.EnvironmentArgumentValue[List[str]]):
+        argument_type = List[str]
+        environment_variable_key = "APP_STORE_CONNECT_DEVICE_UDIDS"
+        example_value = "00000000-000000000000001E"
+
+        def _apply_type(self, non_typed_value: str) -> List[str]:
+            udids = [udid.strip() for udid in shlex.split(non_typed_value) if udid.strip()]
+            if not udids or not all(udids):
+                raise argparse.ArgumentTypeError(f'Provided value "{non_typed_value}" is not valid')
+            return udids
```

Comment:
Splitting should be applied only for arguments that originate from file or environment variable. 
Consider this:
```shell
app-store-connect register-device -u "first udid" "another udid" -n "device-name" --ignore-registration-errors
```
I'd expect that this invocation attempts to register two UDIDs: `first udid` and `another udid`. In practice it attempts to register four UDIDs: `first`, `udid`, `another` and again `udid`. 

Note that when providing the values via environment variable, it is all good.
```shell
APP_STORE_CONNECT_DEVICE_UDIDS="udid1 udid2 'udid 3'" app-store-connect register-device -n "device-name" --ignore-registration-errors
```
Running the above indeed attempts to register three UDIDs: `udid1`, `udid2` and `udid 3` (note the whitespace).

To overcome this you should check the origin of the argument:
```python
is_from_cli = not (self._is_from_environment() or self._is_from_file() or self._from_environment)
```
and apply splitting only in cases if the argument is not provided directly via CLI.

Note that as of now the above check will cause an error as of now since `_from_environment` is defined as the last thing in the `__init__` method of `TypedCliArgument`. This can be easily fixed though.